controllers/machinehealthcheck_controller.go

```diff
@@ -205,6 +206,7 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for i, t := range targets {
 		m.Status.Targets[i] = t.Machine.Name
 	}
+	sort.Strings(m.Status.Targets)
```

Comment:
Can't say I'm a huge fan of requiring sorting here just for the sake of the tests. Would it make sense to create a custom matcher to handle the test for equality instead?